src/bayesian_linear_regression.jl

```diff
@@ -73,3 +73,25 @@ function __compute_inference_quantities(fx::FiniteBLR, y::AbstractVector{<:Real}
 
     return Uw, Bt, δy, logpdf_δy, Λεy
 end
+
+# Random function sample generation
+# Following the Random API: https://docs.julialang.org/en/v1/stdlib/Random/#An-optimized-sampler-with-pre-computed-data
+struct BLRFunctionSample{Tw<:AbstractVector}
+    w::Tw
+end
+
+(s::BLRFunctionSample)(X::AbstractMatrix{<:Real}) = X's.w
+(s::BLRFunctionSample)(X::ColVecs) = X.X's.w
+(s::BLRFunctionSample)(X::RowVecs) = X.X * s.w
+
+Random.eltype(::Type{<:BayesianLinearRegressor}) = BLRFunctionSample
+
+function Random.Sampler(::Type{<:AbstractRNG}, blr::BayesianLinearRegressor, ::Random.Repetition)
+    Λw_U = _cholesky(blr.Λw).U
+    Random.SamplerSimple(blr, (Λw_U=Λw_U, mw=blr.mw))
+end
+
+function Random.rand(rng::AbstractRNG, sp::Random.SamplerSimple{<:BayesianLinearRegressor})
```

Comment:
Ah, yes should be fairly easy

Reply:
Could also add a way to sample a single function with N ouputs (instead of a vector of N functions as currently) - would be much faster for large N for the same reason as this

Reply:
> Ah, yes should be fairly easy

Ace.

> Could also add a way to sample a single function with N ouputs (instead of a vector of N functions as currently) - would be much faster for large N for the same reason as this

What kind of thing do you have in mind here? Are you talking about multi-output BLR, or just evaluating multiple samples from the same BLR quickly?

Reply:
I mean evaluating multiple samples from a single-output BLR more efficiently

Reply:
> Ah, yes should be fairly easy

oh, it's actually not that easy because the multisample version can ask for an arbitrary number of dimensions

Reply:
> oh, it's actually not that easy because the multisample version can ask for an arbitrary number of dimensions

Oh, really? Could you elaborate?

Reply:
added the multisample version